src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ApiExtensions.cs

```diff
@@ -373,5 +383,298 @@ private static void ValidateRequest(string name, string path, string apiVersion,
+                }
+
+                documentationContent.Headers.ContentDisposition = contentDispositionHeaderValue;
+                multiPartContent.Add(documentationContent, "documentation");
+            }
+
+            if (status != null)
+            {
+                StringContent statusContent = new StringContent(SafeJsonConvert.SerializeObject(status, this.SerializationSettings).Trim('"'), System.Text.Encoding.UTF8);
+                multiPartContent.Add(statusContent, "status");
+            }
+
+            httpRequest.Content = multiPartContent;
+
+            // Set Credentials
+            if (this.Credentials != null)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                await this.Credentials.ProcessHttpRequestAsync(httpRequest, cancellationToken).ConfigureAwait(false);
+            }
+
+            // Send Request
+            if (shouldTrace)
+            {
+                ServiceClientTracing.SendRequest(invocationId, httpRequest);
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+            httpResponse = await this.HttpClient.SendAsync(httpRequest, cancellationToken).ConfigureAwait(false);
+            if (shouldTrace)
+            {
+                ServiceClientTracing.ReceiveResponse(invocationId, httpResponse);
+            }
+
+            HttpStatusCode statusCode = httpResponse.StatusCode;
+            cancellationToken.ThrowIfCancellationRequested();
+            string responseContent = null;
+
+            if ((int)statusCode != 200 && (int)statusCode != 400 && (int)statusCode != 404 && (int)statusCode != 409 && (int)statusCode != 422)
+            {
+                var ex = new HttpOperationException(string.Format("Operation returned an invalid status code '{0}'", statusCode));
+                if (httpResponse.Content != null)
+                {
+                    responseContent = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+                }
+                else
+                {
+                    responseContent = string.Empty;
+                }
+
+                ex.Request = new HttpRequestMessageWrapper(httpRequest, requestContent);
+                ex.Response = new HttpResponseMessageWrapper(httpResponse, responseContent);
+                if (shouldTrace)
+                {
+                    ServiceClientTracing.Error(invocationId, ex);
+                }
+
+                httpRequest.Dispose();
+                if (httpResponse != null)
+                {
+                    httpResponse.Dispose();
+                }
+
+                throw ex;
+            }
+
+            // Create Result
+            var result = new HttpOperationResponse<object>();
+            result.Request = httpRequest;
+            result.Response = httpResponse;
+
+            // Deserialize Response
+            if ((int)statusCode == 200)
```

Comment:
Though I would not worry too much about this file which is more of a stop gap arrangement, these IF blocks are duplicated in both create/update. At least in create similarly treated status codes are clubbed into one IF block.  Refactoring can help reduce at least  50-70 lines of code

Reply:
Or, may be we can just ignore it since this file will be eventually removed